pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -140,33 +140,42 @@ func (c ServiceCatalogAPIServerOperator) sync() error {
 		}
 		return nil
 	case operatorsv1.Removed:
-		// TODO probably need to watch until the NS is really gone
-		if err := c.kubeClient.CoreV1().Namespaces().Delete(operatorclient.TargetNamespaceName, nil); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		originalOperatorConfig := operatorConfig.DeepCopy()
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-			Type:    operatorsv1.OperatorStatusTypeAvailable,
-			Status:  operatorsv1.ConditionTrue,
-			Reason:  "Removed",
-			Message: "the apiserver is in the desired state (Removed).",
-		})
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-			Type:    operatorsv1.OperatorStatusTypeProgressing,
-			Status:  operatorsv1.ConditionFalse,
-			Reason:  "Removed",
-			Message: "",
-		})
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-			Type:    operatorsv1.OperatorStatusTypeDegraded,
-			Status:  operatorsv1.ConditionFalse,
-			Reason:  "Removed",
-			Message: "",
-		})
-
-		if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-			if _, err := c.operatorConfigClient.ServiceCatalogAPIServers().UpdateStatus(operatorConfig); err != nil {
-				return err
+		// 4.1 GA TAKE NOTICE -- don't actually take any action to uninstall.  There are no good options since we don't delete the
+		// API Service and we don't want to because of secrets with ownerref & GC - see BZ https://bugzilla.redhat.com/show_bug.cgi?id=1711043
+		// so don't delete the namespace.  If Admins really want to remove Service Catalog they will read the doc that says to contact support
+		// (or contact support directly) and be talked through the necessity to understand the complexities of removing the aggregated API service.
+		// This all will be addressed in 4.1.z ASAP.
+		//
+		if _, err := c.kubeClient.CoreV1().Namespaces().Get(operatorclient.TargetNamespaceName, metav1.GetOptions{}); err == nil {
```

Comment:
sorry i'm getting here late, but this is a little loose as a check...  you could get an err on this Get() for any number of reasons (e.g. network or api server hiccup) and you're defaulting to assuming it's ok to proceed w/ the removal if you get any error.  Ideally i think you should be confirming this err is a NotFound error before assuming it's safe to proceed with a normal removal.

Reply:
We don't actually do any removal now, so it will just mean we don't report the "removal ignored" status. So there is no serious risk, just imperfect failure reporting.

Reply:
i see now, thanks